File: bindings/lua/Polycode/ConfigEntry.lua
class "ConfigEntry"


function ConfigEntry:__getvar(name)
	if name == "key" then
		return Polycode.ConfigEntry_get_key(self.__ptr)
	elseif name == "configNamespace" then
		return Polycode.ConfigEntry_get_configNamespace(self.__ptr)
	elseif name == "numVal" then
		return Polycode.ConfigEntry_get_numVal(self.__ptr)
	elseif name == "stringVal" then
		return Polycode.ConfigEntry_get_stringVal(self.__ptr)
	elseif name == "isString" then
		return Polycode.ConfigEntry_get_isString(self.__ptr)
	end
end

function ConfigEntry:__setvar(name,value)
	if name == "key" then
		Polycode.ConfigEntry_set_key(self.__ptr, value)
		return true
	elseif name == "configNamespace" then
		Polycode.ConfigEntry_set_configNamespace(self.__ptr, value)
		return true
	elseif name == "numVal" then
		Polycode.ConfigEntry_set_numVal(self.__ptr, value)
		return true
	elseif name == "stringVal" then
		Polycode.ConfigEntry_set_stringVal(self.__ptr, value)
		return true
	elseif name == "isString" then
		Polycode.ConfigEntry_set_isString(self.__ptr, value)
		return true
	end
	return false
end
function ConfigEntry:__delete()
	if self then Polycode.delete_ConfigEntry(self.__ptr) end
end

File: bindings/javascript/Polycode/ConfigEntry.js
function ConfigEntry() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.ConfigEntry()
	}
	Object.defineProperties(this, {
		'key': { enumerable: true, configurable: true, get: ConfigEntry.prototype.__get_key, set: ConfigEntry.prototype.__set_key},
		'configNamespace': { enumerable: true, configurable: true, get: ConfigEntry.prototype.__get_configNamespace, set: ConfigEntry.prototype.__set_configNamespace},
		'numVal': { enumerable: true, configurable: true, get: ConfigEntry.prototype.__get_numVal, set: ConfigEntry.prototype.__set_numVal},
		'stringVal': { enumerable: true, configurable: true, get: ConfigEntry.prototype.__get_stringVal, set: ConfigEntry.prototype.__set_stringVal},
		'isString': { enumerable: true, configurable: true, get: ConfigEntry.prototype.__get_isString, set: ConfigEntry.prototype.__set_isString}
	})
}

ConfigEntry.prototype.__get_key = function() {
	return Polycode.ConfigEntry__get_key(this.__ptr)
}

ConfigEntry.prototype.__set_key = function(val) {
	Polycode.ConfigEntry__set_key(this.__ptr, val)
}

ConfigEntry.prototype.__get_configNamespace = function() {
	return Polycode.ConfigEntry__get_configNamespace(this.__ptr)
}

ConfigEntry.prototype.__set_configNamespace = function(val) {
	Polycode.ConfigEntry__set_configNamespace(this.__ptr, val)
}

ConfigEntry.prototype.__get_numVal = function() {
	return Polycode.ConfigEntry__get_numVal(this.__ptr)
}

ConfigEntry.prototype.__set_numVal = function(val) {
	Polycode.ConfigEntry__set_numVal(this.__ptr, val)
}

ConfigEntry.prototype.__get_stringVal = function() {
	return Polycode.ConfigEntry__get_stringVal(this.__ptr)
}

ConfigEntry.prototype.__set_stringVal = function(val) {
	Polycode.ConfigEntry__set_stringVal(this.__ptr, val)
}

ConfigEntry.prototype.__get_isString = function() {
	return Polycode.ConfigEntry__get_isString(this.__ptr)
}

ConfigEntry.prototype.__set_isString = function(val) {
	Polycode.ConfigEntry__set_isString(this.__ptr, val)
}

Duktape.fin(ConfigEntry.prototype, function (x) {
	if (x === ConfigEntry.prototype) {
		return;
	}
	Polycode.ConfigEntry__delete(x.__ptr)
})
